.github/workflows/release.yaml

```
@@ -47,56 +47,110 @@ jobs:
+            echo "No src artifact found matching pattern: $COMMIT_SHA-code-editor-sagemaker-server-src"
             exit 1
           fi
 
       - name: Check artifacts exist and rename
         run: |
           ls -la
-          TARBALL=$(ls *.tar.gz | head -1)
-          if [ -z "$TARBALL" ]; then
+          
+          # Rename build artifact
+          BUILD_TARBALL=$(ls *build*.tar.gz | head -1 || true)
+          if [ -n "$BUILD_TARBALL" ]; then
+            mv "$BUILD_TARBALL" "code-editor-sagemaker-${VERSION_NUM}.tar.gz"
+            echo "Renamed $BUILD_TARBALL to code-editor-sagemaker-${VERSION_NUM}.tar.gz"
+          else
             echo "Error: No build artifacts found for commit $COMMIT_SHA"
             exit 1
           fi
-          mv "$TARBALL" "code-editor-sagemaker-${VERSION_NUM}.tar.gz"
-          echo "Renamed $TARBALL to code-editor-sagemaker-${VERSION_NUM}.tar.gz"
+          
+          # Rename src artifact
+          SRC_TARBALL=$(ls *src*.tar.gz | head -1 || true)
+          if [ -n "$SRC_TARBALL" ]; then
+            mv "$SRC_TARBALL" "code-editor-sagemaker-src-${VERSION_NUM}.tar.gz"
+            echo "Renamed $SRC_TARBALL to code-editor-sagemaker-src-${VERSION_NUM}.tar.gz"
+          else
+            echo "Error: No src artifacts found for commit $COMMIT_SHA"
+            exit 1
+          fi
+      
+      - name: Create tag if not exists
+        run: |
+          # Check if tag already exists
+          if git rev-parse "$VERSION_NUM" >/dev/null 2>&1; then
+            echo "Tag $VERSION_NUM already exists"
+          else
+            echo "Creating tag $VERSION_NUM"
+            git tag "$VERSION_NUM" "$COMMIT_SHA"
+            git push origin "$VERSION_NUM"
+          fi
       
       - name: Create GitHub release
         run: |
-          # Create the release
-          RELEASE_RESPONSE=$(curl -L \
-            -X POST \
+          # Check if release already exists
+          EXISTING_RELEASE=$(curl -s -L \
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer $GH_TOKEN" \
             -H "X-GitHub-Api-Version: 2022-11-28" \
-            "https://api.github.com/repos/$REPOSITORY/releases" \
-            -d "{
-              \"tag_name\": \"$VERSION_NUM\",
-              \"name\": \"Release $VERSION_NUM\",
-              \"draft\": false,
-              \"prerelease\": false
-            }")
-          
-          echo "Release API response: $RELEASE_RESPONSE"
+            "https://api.github.com/repos/$REPOSITORY/releases/tags/$VERSION_NUM" 2>/dev/null)
           
-          if echo "$RELEASE_RESPONSE" | jq -e '.message' > /dev/null; then
-            echo "Error creating release: $(echo "$RELEASE_RESPONSE" | jq -r '.message')"
-            exit 1
+          if echo "$EXISTING_RELEASE" | jq -e '.id' > /dev/null 2>&1; then
+            echo "Release for tag $VERSION_NUM already exists, skipping creation..."
+            RELEASE_RESPONSE="$EXISTING_RELEASE"
```

Nit: same as above, can be more concise with GitHub CLI:
```suggestion
           gh release list "$VERSION_NUM"
```